Entity Passengers get pushed wrongly

SinanAkkoyun opened the pull request:
```js
originalVehicle.passengers = originalVehicle.passengers.splice(index, 1)
```
Hey! If I understood this correctly: Reassignment is unnecessary, splice already modifies the array itself, which resulted in https://github.com/PrismarineJS/mineflayer/issues/3600

extremeheat commented:
It's an unnecessary operation, but unless the old reference is being incorrectly used somewhere over access via `bot`, it's not a bug that should break anything.

rom1504 commented:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/splice#return_value looks like splice returns an array of removed elements so indeed this is a behavior change

rom1504 commented:
Can you please fix the failing tests ?

SinanAkkoyun commented:
> Can you please fix the failing tests ?

I fixed the tests now, I believe my PR was not the issue
The original code `if(vehicle !== null)` did not test for vehicle being `undefined`; I replaced it with `if(vehicle)`

I don't understand the new test fail for 1.13.2:
```
54 passing (4m)
  1 failing

  1) mineflayer_external 1.13.2v
       useChests:

      AssertionError [ERR_ASSERTION]: Expected values to be strictly equal:

27 !== 305

      + expected - actual

      -27
      +305
      
      at /home/runner/work/mineflayer/mineflayer/test/externalTests/useChests.js:86:10
      at process.processTicksAndRejections (node:internal/process/task_queues:105:5)
```

I think this isn't due to my PR, however please let me know if that's the case and I will try to investigate
